gnomad_qc/v4/variant_qc/random_forest.py

```diff
@@ -0,0 +1,353 @@
+
+FEATURES = [
+    "allele_type",
+    "AS_MQRankSum",
+    "AS_pab_max",
+    "AS_QD",
+    "AS_ReadPosRankSum",
+    "AS_SOR",
+    "n_alt_alleles",
+    "variant_type",
+]
+INFO_FEATURES = [
+    "AS_MQRankSum",
+    "AS_pab_max",
+    "AS_QD",
+    "AS_ReadPosRankSum",
+    "AS_SOR",
+]
+LABEL_COL = "rf_label"
+PREDICTION_COL = "rf_prediction"
+TRAIN_COL = "rf_train"
+TRUTH_DATA = ["hapmap", "omni", "mills", "kgp_phase1_hc"]
+
+def train_rf(
+    ht: hl.Table,
+    fp_to_tp: float = 1.0,
+    num_trees: int = 500,
+    max_depth: int = 5,
+    no_transmitted_singletons: bool = False,
+    no_sibling_singletons: bool = False,
+    filter_centromere_telomere: bool = False,
+    test_intervals: Union[str, List[str]] = "chr20",
+):
+    """
+    Train random forest model using `train_rf_model`.
+
+    :param ht: Table containing annotations needed for RF training, built with ???
+    :param fp_to_tp: Ratio of FPs to TPs for creating the RF model. If set to 0, all training examples are used.
+    :param num_trees: Number of trees in the RF model.
+    :param max_depth: Maxmimum tree depth in the RF model.
+    :param no_transmitted_singletons: Do not use transmitted singletons for training.
+    :param no_sibling_singletons: Do not use sibling singletons for training. 
+    :param filter_centromere_telomere: Filter centromeres and telomeres before training.
+    :param test_intervals: Specified interval(s) will be held out for testing and evaluation only. (default to "chr20")
+    :return: `ht` annotated with training information and the RF model
+    """
+    features = FEATURES
+    test_intervals = test_intervals
+
+    fp_expr = ht.fail_hard_filters
+    tp_expr = ht.omni | ht.mills
+    if not no_transmitted_singletons:
+        tp_expr = tp_expr | ht.transmitted_singleton 
+    if not no_sibling_singletons:
+        tp_expr = tp_expr | ht.sibling_singleton # TODO: make sure this is acceptable Hail logic
+
+    if test_intervals:
+        if isinstance(test_intervals, str):
+            test_intervals = [test_intervals]
+        test_intervals = [
+            hl.parse_locus_interval(x, reference_genome="GRCh38")
+            for x in test_intervals
+        ]
+
+    ht = ht.annotate(tp=tp_expr, fp=fp_expr)
+
+    if filter_centromere_telomere:
+        logger.info("Filtering centromeres and telomeres from HT...")
+        rf_ht = ht.filter(~hl.is_defined(telomeres_and_centromeres.ht()[ht.locus]))
+    else:
+        rf_ht = ht
+
+    rf_ht, rf_model = train_rf_model(
+        rf_ht,
+        rf_features=features,
+        tp_expr=rf_ht.tp,
+        fp_expr=rf_ht.fp,
+        fp_to_tp=fp_to_tp,
+        num_trees=num_trees,
+        max_depth=max_depth,
+        test_expr=hl.literal(test_intervals).any(
+            lambda interval: interval.contains(rf_ht.locus)
+        ),
+    )
+
+    logger.info("Joining original RF Table with training information")
+    ht = ht.join(rf_ht, how="left")
+
+    return ht, rf_model
+
+
```

Comment:
Change to use resources like in other scripts. For example: https://github.com/broadinstitute/gnomad_qc/blob/0c52cf47e48fa5b503d874e96482ea4286474c71/gnomad_qc/v4/annotations/generate_variant_qc_annotations.py#L531

https://github.com/broadinstitute/gnomad_qc/blob/0c52cf47e48fa5b503d874e96482ea4286474c71/gnomad_qc/v4/sample_qc/identify_trios.py#L387

and others in v4 have this structure